### PR TITLE
fix empty vertx server metrics with http2

### PR DESCRIPTION
**Pull request by jo5ef**
I've described the problem in issue #2749. This is my temporary quick fix that works for us for now, but I haven't changed/added tests and don't have a deep understanding of the new vertx http code in general.

### CODE REVIEW

File: src/test/java/io/vertx/core/http/HttpMetricsTestBase.java [hunk 1]
```diff
@@ -227,6 +228,7 @@ public void testClientConnectionClosed() throws Exception {
   }
 
   @Test
+  @Ignore
```
Comment (vietj) [inline src/test/java/io/vertx/core/http/HttpMetricsTestBase.java]: why is that necessary ?

**Comment by vietj** [comment]
can you add a metrics test too ?

**Comment by jo5ef** [comment]
I think the existing tests are good, I re-enabled & fixed the one that was failing due to my change.

File: src/test/java/io/vertx/core/http/HttpMetricsTestBase.java (hunk 1 above)
Comment (jo5ef) [reply src/test/java/io/vertx/core/http/HttpMetricsTestBase.java]: re-enabled & fixed now, see latest commit

**Comment by vietj** [comment]
I don't see what should have failed in the tests that your patch fixes, can you elaborate ?

**Comment by jo5ef** [comment]
The `HttpServerResponse.closeHandler` behaves differently for HTTP/1.x and HTTP/2. For HTTP/1.x it is called when the connection is closed, for HTTP/2 it is called when the stream is closed, i.e. it is called for *every* request.

The test uses a `closeHandler` to assert hat the metric has `failed`, which is correct for HTTP/1.x. For HTTP/2 however this is not correct, otherwise there would never be any metrics (reason for my issue & PR). Since this test is in a base class for both `Http2MetricsTest` and `Http1xMetricsTest`, it failed for HTTP/2.

I made another change to improve the test to account for both cases.

**Comment by jo5ef** [comment]
Works again after the changes in the http code. Tested with 3.6.3. Thanks!
